### PR TITLE
Merge Dev into Beta

### PR DESCRIPTION
**Pull request by BadWolf42**
Changelog & Typo
Cosmestic code change
Display orphan commands first (if any)
Highlight value change in Command View in real time
Move Add equipment to top 
Adapt js to ask for Broker at eq creation
Add an ajax function to list Brokers
Remove Inclusion Mode
Move brokers in their planels
Highlight brokers in yellow in their panels
Implement Tableview
Prepare jMQTT for "Auto Discovery" and "Real Time" modals
Disable Mosquitto installation by default
Use const for templates path 
Move mousedown on Eq save hack in saveEqLogic(_eqLogic) function 
Add comments in jMQTT.js
Rename id #dTls* -> #jmqttDivTls*
Change Broker Configuration UI to look like MQTT Manager configuration
Remove Certificate Upload/Delete system
Fix some PATH_TEMPLATES_*
Fix an issue with Battery convertion
Move certificat to Database 
Remove mqttTls config and use mqttProto (for mqtt/mqtts and soon ws/wss)
Adapt daemon to handle certificates directly as string
Fix rounded design in Broker tab
Rename Shelly Plus 1pm template
Remove websocket-client from venv requirements
Add system dependant information in dep install
Add last supported version warning on Jeedom 3.3
Revert commit 2aa6756 (Jeedom 3.3 compatibility)
Add more debug info in jMQTT::dependancy_info()
Implement feature request from issue #208: In case of an action cmd, if Value is empty, then tag corresponding to the name of the action is silently used: slider -> #slider#, message -> #message#, color -> #color#, select -> #select# & other -> (vide)
Use background action for listeners (default)
Move CONF_KEY_JSON_PATH in jMQTTCmd and use cmd->get/setJsonPath() when possible
Add CONF_KEY_AUTOPUB in jMQTTCmd and use it when possible
Add CONF_KEY_QOS in jMQTTCmd and use it when possible
Add CONF_KEY_IRREMOVABLE in jMQTTCmd and use it when possible
Add CONF_KEY_REQUEST in jMQTTCmd and use it when possible
Add CONF_KEY_RETAIN in jMQTTCmd and use it when possible
Don't blur certificates textarea on focus
Documentation update
Bump galbar/jsonpath
Fix CONF_KEY_JSON_PATH in jMQTTCmd
Remove brk/eq fine status from Health modale 
Add Modify/Delete Eq in Health modale
Small cleanup of removeJmqtt callback
Add eqLogic status (from health modal) on main page in table view
Rename rf433 node image and remove old ones
Cleanup in jMQTT.js
Send icone list dynamically
Cleanup status icons in tableView
Update debug modale
Cleanup health modal
Cleanup Jeedom 3 warning
Fix Log icon on Broker tab
Compute icons on main page on client-side (js)
Cleanup JS on update Broker status
TODOs in JS
Remove 2x EventState handling...
Add Template Zwavejs2mqtt Fibaro FGRGBW-442 RGBW Controller 2
Fixme Commands used in templates are not converted on template import/export
Template Zwavejs2mqtt Fibaro FGMS-001 Motion Sensor
Changelog

### CODE REVIEW

File: desktop/php/jMQTT_broker.php [hunk 1]
```diff
@@ -75,79 +75,93 @@
...
-							<label class="col-lg-4 control-label">{{IP/Nom de Domaine du Broker}} <sup><i class="fa fa-question-circle tooltips"
-							title="{{IP/Nom de Domaine du Broker, par défaut 'localhost' i.e. la machine hébergeant Jeedom.}}"></i></sup></label>
-							<div class="col-lg-4">
-								<input class="eqLogicAttr form-control" data-l1key="configuration" data-l2key="mqttAddress"
-									style="margin-top: 5px" placeholder="localhost" />
+							<label class="col-lg-4 control-label">{{Adresse du broker}} <sup><i class="fa fa-question-circle tooltips" title="{{Paramètres d'accès au Broker.}}"></i></sup></label>
+							<div class="col-lg-7 input-group">
+								<span class="input-group-btn">
+									<select class="eqLogicAttr form-control tooltips" data-l1key="configuration" data-l2key="mqttProto" style="width:80px;"
+										title="{{Choisir si le Broker attend une communication sécurité.<br />Pour plus d'information, se référer à la documentation.}}">
```
Comment (Domochip) [inline desktop/php/jMQTT_broker.php:87]: sécurité -> sécurisée

Comment (BadWolf42) [reply desktop/php/jMQTT_broker.php:87]: zoups, fixed in next beta

File: plugin_info/install.php [hunk 2]
```diff
@@ -262,6 +262,33 @@ function convertBatteryStatus() {
...
+			$broker->setConfiguration(jMQTT::CONF_KEY_MQTT_TLS_CA, file_get_contents($oldPath.'/'.$cert));
+		$cert = $broker->getConfiguration('mqttTlsClientCertFile', '');
+		$broker->setConfiguration('mqttTlsClientCertFile', null);
+		if ($cert != '')
+			$broker->setConfiguration(jMQTT::CONF_KEY_MQTT_TLS_CLI_CERT, file_get_contents($oldPath.'/'.$cert));
+		$cert = $broker->getConfiguration('mqttTlsClientKeyFile', '');
+		$broker->setConfiguration('mqttTlsClientKeyFile', null);
+		if ($cert != '')
+			$broker->setConfiguration(jMQTT::CONF_KEY_MQTT_TLS_CLI_KEY, file_get_contents($oldPath.'/'.$cert));
+		$broker->save();
```
Comment (Domochip) [inline plugin_info/install.php:283]: good idea for the migration, but there is a potential issue : certificate files can be binary encoded (DER) and so getting content and put it in configuration won't work.

File: docs/fr_FR/index.md [hunk 3]
```diff
@@ -32,39 +32,58 @@ Les géants du web parmi lesquels AWS ou Microsoft utilisent MQTT pour remonter
...
-C'est aussi sur cette page qu'il est possible d'ajouter et de supprimer des certificats SSL pour les Broker.
+jMQTT est un Client MQTT pour Jeedom, il faut donc un Broker pour pouvoir d'utiliser.
+
+Par défaut, jMQTT installe et configure le Broker "Mosquitto" sur la machine hébergeant Jeedom pendant l'installation des dépendances.
+Il suffit ensuite de configurer vos modules domotique compatible MQTT pour qu'ils se connectent au Broker local (IP de votre Jeedom).
+
+> **Important**
+>
+> Si vous avez déjà un Broker ou que vous utilisez aussi un autre plugin MQTT qui a déjà installé un Broker, désactivez l'installation de Mosquitto et configurez l'autre Broker dans jMQTT.
+> Pour ne pas installer le Broker Mosquitto sur la machine hébergeant Jeedom, il faut décocher la case *Installer Mosquitto localement* et sauvegarder la configuration.
```
Comment (Domochip) [inline docs/fr_FR/index.md]: not installed by default anymore :-)

File: desktop/php/jMQTT_eqpt.php [hunk 4]
```diff
@@ -125,57 +125,18 @@
...
+				<script>
+					// Initialize Icon dropbox from jmqttIcons global var
+					// This MUST be done here, only once and before loading plugin.template.js
+					var icos = $('.eqLogicAttr[data-l1key=configuration][data-l2key=icone]');
+					icos.html('');
+					$.each(jmqttIcons, function(key) {
+						opt = new Option(jmqttIcons[key]['name'], jmqttIcons[key]['id']);
+						opt.setAttribute('file', jmqttIcons[key]['file']);
+						icos.append(opt);
+					});
```
Comment (Domochip) [inline desktop/php/jMQTT_eqpt.php:138]: What about icon names translation?

File: core/config/template/Zwavejs2mqtt_Fibaro_FGRGBW-442_RGBW_Controller_2.json [hunk 5]
```diff
@@ -0,0 +1,806 @@
+{
+    "Zwavejs2mqtt Fibaro FGRGBW-442 RGBW Controller 2": {
```
Comment (Domochip) [inline core/config/template/Zwavejs2mqtt_Fibaro_FGRGBW-442_RGBW_Controller_2.json:2]: Zwavejs2mqtt has been renamed [zwave-js-ui](https://github.com/zwave-js/zwave-js-ui) :-(

File: desktop/php/jMQTT_eqpt.php (hunk 4 above)
Comment (BadWolf42) [reply desktop/php/jMQTT_eqpt.php:138]: Yes, that's the point of this change, it is done server side here:
https://github.com/Domochip/jMQTT/blob/330d3a1cfefacc4edce75cc9270f5a8241a94841/desktop/php/jMQTT.php#L63-L70

File: core/config/template/Zwavejs2mqtt_Fibaro_FGRGBW-442_RGBW_Controller_2.json (hunk 5 above)
Comment (BadWolf42) [reply core/config/template/Zwavejs2mqtt_Fibaro_FGRGBW-442_RGBW_Controller_2.json:2]: Yes I saw that a few days ago. 
I'll think about renaming everything for the next beta.
A little grep/sed will certainly do the trick.

File: plugin_info/install.php (hunk 2 above)
Comment (BadWolf42) [reply plugin_info/install.php:283]: This should not be an issue as Paho MQTT only wants PEM files:
https://github.com/eclipse/paho.mqtt.python/blob/9782ab81fe7ee3a05e74c7f3e1d03d5611ea4be4/src/paho/mqtt/client.py#L750-L751
(That's why I took the liberty to directly transfert files content from config)

Comment (BadWolf42) [reply plugin_info/install.php:283]: Also I think that's a good way to ensure certificates are provided in PEM format by users.

Comment (Domochip) [reply plugin_info/install.php:283]: ok, perfect

**Pull request merged**